File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {

    void inorder_traversal(TreeNode* root, vector<int>& arr)
    {
        if (!root)
            return ;
        inorder_traversal(root->left, arr);
        arr.push_back(root->val);
        inorder_traversal(root->right, arr);
    }

public:
    int kthSmallest(TreeNode* root, int k) {
        // since  this binary tree we can get its values first.
        vector<int>arr;
        inorder_traversal(root, arr);
        return (arr[k - 1]);
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        level = 0
        queue = deque()
        current = root
        if root:
            queue.append(root)
        while len(queue) > 0:
            size = len(queue)
            for i in range(size):
                current = queue[0]
                queue.popleft()
                if current.left:
                    queue.append(current.left)
                if current.right:
                    queue.append(current.right)
            level += 1
        return level